### ISSUE TITLE
Support for Spring Boot 3.0

### ISSUE THREAD

**Issue by MiguelAngelLV** [turkraft/spring-filter#243]
When I try use the library in Spring Boot 3.0, I get error from jakarta.

```
java.lang.AbstractMethodError: Receiver class com.turkraft.springfilter.boot.FilterSpecification does not define or inherit an implementation of the resolved method 'abstract jakarta.persistence.criteria.Predicate toPredicate(jakarta.persistence.criteria.Root, jakarta.persistence.criteria.CriteriaQuery, jakarta.persistence.criteria.CriteriaBuilder)' of interface org.springframework.data.jpa.domain.Specification.

```

The actual version implement `javax.persistance` instead of `jakarta.persistance`

**Comment by pkukharenka** [comment]
+1

**Comment by MiguelAngelLV** [comment]
I am trying to convert the project to SB 3.0.0, but I found one problem: PluralAttributePath is not available in Hibernate 6.1...

**Comment by MiguelAngelLV** [comment]
Any ideas?

**Comment by candogusyilmaz** [comment]
+1

**Comment by sisimomo** [comment]
That would be really cool! This project is the only dependency in my project that does not support Spring Boot 3.0 :( .

@torshid if you have a few minutes, it would be really nice of you to help use support Spring Boot 3.0 :) .

Thanks! For your already amazing work!

**Comment by marcopag90** [comment]
Any news on this?

**Comment by tnoroozi** [comment]
i have same problem. 
@torshid have you any plan to upgrade it ?

@sisimomo do you have any success to upgrade it ?

**Comment by pkukharenka** [comment]
Seems like this project is dead. 

I refactored and adopted it to spring boot 3 myself. Can't share the whole adoption code since I removed all the logic related to document filtering.

**Comment by torshid** [comment]
Hi,

The project is **not** dead, that is wrong. I am working on a new version which will be published soon, adding a lot of new features, with support for Spring Boot 3. Many issues will be closed.

I've been working on this new release for a month now. Consider supporting the project by donating, that would be a great encouragement (specifically addressing the multiple enterprises using this project).

Thanks.

**Comment by torshid** [comment]
I have released a snapshot of release 3.0.0 which supports Spring Boot 3, see https://github.com/turkraft/spring-filter/issues/254.
Feel free to open issues if you see anything wrong.

**Comment by marcopag90** [comment]
Thanks. As usual, amazing work. I'm going to try and integrate it inside a personal open source project and see how it works :)
